lib/service/service.go

```diff
@@ -555,6 +555,10 @@ func NewTeleport(cfg *Config) (*TeleportProcess, error) {
 		}
 	}
 
+	if uuid.Parse(cfg.HostUUID) == nil {
+		log.Warnf("Host UUID %q is not a true UUID (may interfere with some functionality)", cfg.HostUUID)
```

Comment:
I would explain what could happen if the value in not unique.

```go
log.Warnf("Host UUID %q is not a valid UUID. If the Host UUID value in not unique, this may lead to clients inadvertently connecting to the wrong node.", cfg.HostUUID)
```